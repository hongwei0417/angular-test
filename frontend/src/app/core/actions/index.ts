import * as PageActions from './page.actions';
import * as HttpActions from './http.actions';

export { PageActions, HttpActions };
