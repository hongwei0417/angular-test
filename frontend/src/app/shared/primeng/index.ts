// module
export * from './primeng.module';
