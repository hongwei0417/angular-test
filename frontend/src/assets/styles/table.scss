

//#region table
.p-datatable-scrollable-wrapper{
  box-shadow: $shadow-500;
}

//清除預設值
.p-datatable .p-sortable-column .p-sortable-column-icon{
  margin-left: 0;
}

//移除套版預設為滾動條預留的空間
.p-datatable-scrollable-header-box{
  padding-right: 0 !important;
}


.p-sortable-column-icon{
  font-size: 12px;
}

th,td{
  color: $gray-500;
  font-size: 12px;
}



td{
  word-break: break-all;
}

.p-datatable .p-datatable-thead > tr > th{
  color:$gray-500;
  font-weight: 500;
}

.p-datatable .p-datatable-thead > tr > th,
.p-datatable .p-datatable-tbody > tr > td{
  padding: 10px;
}

.p-datatable .p-datatable-thead > tr > th:first-child,
.p-datatable .p-datatable-tbody > tr > td:first-child{
  padding-left: 15px;
}

.p-datatable .p-datatable-thead > tr > th:last-child,
.p-datatable .p-datatable-tbody > tr > td:last-child{
  padding-right: 15px;
}

.pi-sort-alt:before,
.pi-sort-amount-up-alt:before,
.pi-sort-amount-down:before{
  font-family: "Font Awesome 5 Free";
  font-weight: 900;
}

.pi-sort-alt:before{
  content: '\f0dc';
}

.pi-sort-amount-up-alt:before{
  content: '\f0de';
}

.pi-sort-amount-down:before{
  content: '\f0dd';
}

.p-datatable .p-sortable-column:focus{
  box-shadow: none;
}

.p-datatable .p-sortable-column:not(.p-highlight):hover,
.p-datatable .p-sortable-column.p-highlight,
.p-datatable .p-sortable-column.p-highlight:hover{
  background: $gray-200;
  color:$primary-medium;
}


.p-datatable .p-sortable-column .p-sortable-column-icon{
  color: #CED3D3;
}


.p-datatable .p-sortable-column.p-highlight .p-sortable-column-icon,
.p-datatable .p-sortable-column.p-highlight:hover .p-sortable-column-icon,
.p-datatable .p-sortable-column:not(.p-highlight):hover .p-sortable-column-icon{
  color: $primary-medium;
}



// #region  table頁碼 共用區
p-paginator {
  display: flex;
  justify-content: flex-end;
}

.p-datatable .p-paginator-bottom{
  border: none;
}

.p-paginator{
  margin-top: 20px;
  padding: 0;
  background: transparent;
}

.p-paginator .p-paginator-first,
.p-paginator .p-paginator-prev,
.p-paginator .p-paginator-next,
.p-paginator .p-paginator-last,
.p-paginator .p-paginator-pages .p-paginator-page{
  background: transparent;
  border: none;
  color: $gray-500;
}


.p-paginator .p-paginator-pages .p-paginator-page.p-highlight{
  background: $primary-medium;
  border: none;
}



// .p-link{
//   font-size: 18px;
// }

.p-link:focus{
  box-shadow: none;
}

.p-paginator .p-paginator-pages .p-paginator-page:not(.p-highlight):hover,
.p-paginator .p-paginator-first:not(.p-disabled):not(.p-highlight):hover,
.p-paginator .p-paginator-prev:not(.p-disabled):not(.p-highlight):hover,
.p-paginator .p-paginator-next:not(.p-disabled):not(.p-highlight):hover,
.p-paginator .p-paginator-last:not(.p-disabled):not(.p-highlight):hover{
  color: $gray-500;
}
// #endregion  table頁碼 共用區


//#endregion table
