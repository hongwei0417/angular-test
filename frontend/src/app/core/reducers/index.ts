import {
  Action,
  ActionReducer,
  ActionReducerMap,
  createFeatureSelector,
  createSelector,
  MetaReducer,
} from '@ngrx/store';
import * as fromRouter from '@ngrx/router-store';
import { environment } from '../../../environments/environment';
import { InjectionToken } from '@angular/core';

export interface IState {
  router: fromRouter.RouterReducerState;
}

export const reducers: ActionReducerMap<IState> = {
  router: fromRouter.routerReducer,
};

export const ROOT_REDUCERS = new InjectionToken<
  ActionReducerMap<IState, Action>
>('Root reducers token', {
  factory: () => {
    return {
      router: fromRouter.routerReducer,
    };
  },
});

export const metaReducers: MetaReducer<IState>[] = !environment.production
  ? []
  : [];
