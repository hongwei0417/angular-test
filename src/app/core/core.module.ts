import { RouterModule } from '@angular/router';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FakeDataService } from './services/fake-data.service';
import { AppComponent } from './containers/app/app.component';
import { LayoutComponent } from './components/layout/layout.component';

@NgModule({
  declarations: [AppComponent, LayoutComponent],
  imports: [CommonModule, RouterModule],
  exports: [AppComponent, LayoutComponent],
})
export class CoreModule {}
