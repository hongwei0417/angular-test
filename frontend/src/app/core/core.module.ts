import { HttpHelperService } from './services/httpHelper/http-helper.service';
import { CommonModule } from '@angular/common';
import { NgModule, Optional, SkipSelf } from '@angular/core';
import { RouterModule } from '@angular/router';
import { SharedModule } from '../shared';
import { ConfigComponent } from './components/config/config.component';
import { FooterComponent } from './components/footer/footer.component';
import { InlineProfileComponent } from './components/inline-profile/inline-profile.component';
import { MenuComponent } from './components/menu/menu.component';
import { MenuitemComponent } from './components/menuitem/menuitem.component';
import { TopbarComponent } from './components/topbar/topbar.component';
import { AppComponent } from './containers/app/app.component';
import { LayoutComponent } from './containers/layout/layout.component';
import { EnsureModuleLoadedOnceGuard } from './ensure-module-loaded-once.guard';
import { EffectsModule } from '@ngrx/effects';
import { MessageService } from 'primeng/api';
import { httpInterceptorProviders } from './interceptors';

@NgModule({
  declarations: [
    AppComponent,
    LayoutComponent,
    ConfigComponent,
    FooterComponent,
    InlineProfileComponent,
    MenuComponent,
    MenuitemComponent,
    TopbarComponent,
  ],
  imports: [CommonModule, RouterModule, SharedModule],
  exports: [AppComponent],
  providers: [HttpHelperService, MessageService, httpInterceptorProviders],
})
export class CoreModule extends EnsureModuleLoadedOnceGuard {
  constructor(@Optional() @SkipSelf() parentModule: CoreModule) {
    super(parentModule);
  }
}
