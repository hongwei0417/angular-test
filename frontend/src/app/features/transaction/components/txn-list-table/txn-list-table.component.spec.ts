import { By } from '@angular/platform-browser';
import { DropdownModule } from 'primeng/dropdown';
import { CommonModule } from '@angular/common';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { TableModule } from 'primeng/table';
import { MockModule } from 'ng-mocks';

import {
  autoschtranms,
  tableColumns,
  TxnListTableComponent,
} from './txn-list-table.component';

describe('TxnListTableComponent', () => {
  let component: TxnListTableComponent;
  let fixture: ComponentFixture<TxnListTableComponent>;
  let listTableElement: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [CommonModule, ReactiveFormsModule, TableModule],
      declarations: [TxnListTableComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(TxnListTableComponent);
    component = fixture.componentInstance;
    listTableElement = fixture.nativeElement as HTMLElement;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should set table header', () => {
    expect(component.tableCols).toEqual(tableColumns);
  });

  it('should set table data', () => {
    expect(component.tableData).toEqual(autoschtranms);
  });

  it('should display correct table data', () => {
    const element = listTableElement.querySelectorAll(
      '.p-datatable-scrollable-body tr'
    );
    const head = tableColumns.splice(-(tableColumns.length - 5));
    autoschtranms.forEach((rowData: any, i) => {
      const row = element[i] as HTMLTableRowElement;
      head.forEach((headData, j) => {
        const cell = row.cells[j + 5] as HTMLTableCellElement;
        expect(cell.textContent?.trim()).toBe(rowData[headData.field] || '');
      });
    });
  });
});
