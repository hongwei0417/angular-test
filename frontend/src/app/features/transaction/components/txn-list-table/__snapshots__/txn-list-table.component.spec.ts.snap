// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxnListTableComponent should match snapshot 1`] = `
<app-txn-list-table
  __ngContext__={[Function LRootView]}
  fb={[Function FormBuilder]}
  removeEvent={[Function EventEmitter_]}
  searchForm={[Function FormGroup]}
  tableCols={[Function Array]}
  tableData={[Function Array]}
>
  <div
    class="transactionList"
  >
    <div
      class="container"
    >
      <div
        class="title"
      >
        <h2>
          Transaction List
        </h2>
        <button
          class="txtBtn customBtn add"
          routerlink="/transaction/create"
        >
           Create New Transaction 
        </button>
      </div>
      <form
        class="fieldContent ng-untouched ng-pristine ng-valid"
        novalidate=""
      >
        <div
          class="itemGroup"
        >
          <div
            class="item insertIcon filter"
          >
            <label>
              Global Filter
            </label>
            <input
              class="ng-untouched ng-pristine ng-valid"
              formcontrolname="globalFilter"
              placeholder="Search"
              type="text"
            />
          </div>
        </div>
      </form>
      <p-table>
        <div
          class="p-datatable p-component p-datatable-scrollable"
          data-scrollselectors=".p-datatable-scrollable-body, .p-datatable-unfrozen-view .p-datatable-scrollable-body"
        >
          <div
            class="p-datatable-scrollable-wrapper"
          >
            <div
              class="p-datatable-scrollable-view"
              style="width: calc(100% - undefined);"
            >
              <div
                class="p-datatable-scrollable-header"
              >
                <div
                  class="p-datatable-scrollable-header-box"
                  style="padding-right: 0px;"
                >
                  <table
                    class="p-datatable-scrollable-header-table"
                  >
                    <thead
                      class="p-datatable-thead"
                    >
                      <tr>
                        <th>
                           View 
                        </th>
                        <th>
                           Edit 
                        </th>
                        <th>
                           Copy 
                        </th>
                        <th>
                           Execute 
                        </th>
                        <th>
                           Reload 
                        </th>
                        <th
                          aria-sort="none"
                          class="p-sortable-column"
                          role="columnheader"
                          tabindex="0"
                        >
                           Transaction Id
                          <p-sorticon>
                            <i
                              class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                            />
                          </p-sorticon>
                        </th>
                        <th
                          aria-sort="none"
                          class="p-sortable-column"
                          role="columnheader"
                          tabindex="0"
                        >
                           Transaction Name
                          <p-sorticon>
                            <i
                              class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                            />
                          </p-sorticon>
                        </th>
                        <th
                          aria-sort="none"
                          class="p-sortable-column"
                          role="columnheader"
                          tabindex="0"
                        >
                           Start Time
                          <p-sorticon>
                            <i
                              class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                            />
                          </p-sorticon>
                        </th>
                        <th
                          aria-sort="none"
                          class="p-sortable-column"
                          role="columnheader"
                          tabindex="0"
                        >
                           End Time
                          <p-sorticon>
                            <i
                              class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                            />
                          </p-sorticon>
                        </th>
                        <th
                          aria-sort="none"
                          class="p-sortable-column"
                          role="columnheader"
                          tabindex="0"
                        >
                           Active Flag
                          <p-sorticon>
                            <i
                              class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                            />
                          </p-sorticon>
                        </th>
                        <th
                          aria-sort="none"
                          class="p-sortable-column"
                          role="columnheader"
                          tabindex="0"
                        >
                           AP Booking
                          <p-sorticon>
                            <i
                              class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                            />
                          </p-sorticon>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="p-datatable-tbody"
                    >
                    </tbody>
                  </table>
                </div>
              </div>
              <div
                class="p-datatable-scrollable-body"
              >
                <table>
                  <tbody
                    class="p-datatable-tbody"
                  >
                    <tr>
                      <td>
                        <button
                          class="iconBtn customBtnReverse view"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse edit"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse copy"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse execute"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse reload"
                          type="button"
                        />
                      </td>
                      <td>
                         0000000061 
                      </td>
                      <td>
                         A73A_ForPortalCPHoldDisposition 
                      </td>
                      <td>
                         20190814 092000 
                      </td>
                      <td>
                      </td>
                      <td>
                         N 
                      </td>
                      <td>
                         EBAP-HLO999 
                      </td>
                    </tr>
                    <tr>
                      <td>
                        <button
                          class="iconBtn customBtnReverse view"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse edit"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse copy"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse execute"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse reload"
                          type="button"
                        />
                      </td>
                      <td>
                         0000000060 
                      </td>
                      <td>
                         OVT_InvAdjReport_Export_Mail 
                      </td>
                      <td>
                         20200415 000000 
                      </td>
                      <td>
                      </td>
                      <td>
                         N 
                      </td>
                      <td>
                         EBAP-HLO999 
                      </td>
                    </tr>
                    <tr>
                      <td>
                        <button
                          class="iconBtn customBtnReverse view"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse edit"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse copy"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse execute"
                          type="button"
                        />
                      </td>
                      <td>
                        <button
                          class="iconBtn customBtnReverse reload"
                          type="button"
                        />
                      </td>
                      <td>
                         0000000059 
                      </td>
                      <td>
                         OVT_InvAdjusmentReport 
                      </td>
                      <td>
                         20200415 000000 
                      </td>
                      <td>
                      </td>
                      <td>
                         N 
                      </td>
                      <td>
                         EBAP-HLO999 
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div
                class="p-datatable-scrollable-footer"
              >
                <div
                  class="p-datatable-scrollable-footer-box"
                  style="padding-right: 0px;"
                >
                  <table
                    class="p-datatable-scrollable-footer-table"
                  >
                    <tfoot
                      class="p-datatable-tfoot"
                    >
                    </tfoot>
                  </table>
                </div>
              </div>
            </div>
          </div>
          <p-paginator
            styleclass="p-paginator-bottom"
          >
          </p-paginator>
        </div>
      </p-table>
    </div>
  </div>
</app-txn-list-table>
`;
